### ISSUE TITLE
Optionally hide low probability rows

### ISSUE THREAD

**Issue by theRTC204** [mikebryant/ac-nh-turnip-prices#192]
We have seen comments from users that the table is overwhelming at the start of the week, and contains a lot of data that doesn't have a significant amount of value.

We want to retain as much granular information about the predicted rows as possible, so I'm thinking something simple such as the following might be a decent compromise.

- Default behavior shows all rows.
- A toggle switch above the table to hide rows below 1% probability.
- While enabled, show a message at the bottom of the table indicating some rows are hidden.

If a given row has an individual probability of lower than 1%, that row will be hidden from the results. If the overall pattern group ultimately has no remaining results after filtering, that entire group will be removed.

Special care will need to be paid to ensuring that the probabilities (both granular & aggregate) are not changed as a result of the total row count changing.

The risk here I think is that probabilities can swing pretty dramatically as more sell prices are provided. This could lead to patterns which were previously hidden popping up seemingly out of nowhere because they were previously below the "show" threshold.

By keeping the threshold really low at < 1%, we will match the majority of Fluctuating rows, which are the biggest offender of bloating the results, while limiting the amount of unexpected pop-in.

_Originally posted by @theRTC204 in https://github.com/mikebryant/ac-nh-turnip-prices/issues/187#issuecomment-616191906_

**Comment by doopl** [comment]
There is a common situation that produces a large table where this would have no effect.

According to the site's own numbers, 32.9% of weeks will start with a price in the 0.9x-1.4x range. When this happens there is one possible small spike pattern and 42 possible fluctuating pattern variations. Every row has a probability > 1% and so no rows would be culled.

![image](https://user-images.githubusercontent.com/1728488/79692297-353cca80-825c-11ea-9dee-742755f1b390.png)

**Comment by theRTC204** [comment]
At the start of a fresh week, with no data (including an unknown pattern last week) and a buy price of 103, the probability of getting a Fluctuating pattern is aggregated to 34.6%, with 56 possible iterations, but only 5 of them are above 1%. Those numbers don't change much even if you choose a known pattern for last week.
https://turnipprophet.io?prices=103
https://turnipprophet.io?prices=103&pattern=0
https://turnipprophet.io?prices=103&pattern=1
https://turnipprophet.io?prices=103&pattern=2
https://turnipprophet.io?prices=103&pattern=3

Based on the calcuations of the patterns, using this buy price as an example, if you end up with a sell price in Monday AM that is anything 99 or above, you have a 90.2% chance of landing a Fluctuating pattern. At that point, it is not really any longer a question of which pattern you will get, but more which iteration of that pattern - and in that situation, why would you not want to see every possible version, so you can make the best decision.

The fact that the table is 47 rows long here makes little difference, as there is virtually no reason to scroll the table at all. Records are sorted by highest probability, so your 90.2% Fluctuating aggregate is at the top, and it's iterations are also sorted by highest individual probability.

----

Now, all that being said, I do see value in shortening the list when your pattern is less obvious, because in that situation you may want to scroll through the table to see the potential results of the different patterns you might end up with.

**Comment by doopl** [comment]
Thanks for taking the time to think about this.

> if you end up with a sell price in Monday AM that is anything 99 or above, you have a 90.2% chance of landing a Fluctuating pattern. At that point, it is not really any longer a question of which pattern you will get

A user of the site having just entered their data will not yet know the answer to that question. The only way they can currently get it is to look down the entirety of the giant table. At this size the table is hard to read and it is not even clear that the 90.2% is an aggregate probability or what it represents. If the user doesn't already know what they're looking for I think there's a good chance they won't be able to learn anything from the table.

> why would you not want to see every possible version, so you can make the best decision.

Because:

1. You're looking at a giant wall of numbers. I have some limited interest in viewing such things. I think most others have much less.
2. There is no decision to be made at this stage of the week. Assuming you are on the fluctuating pattern, any price could be in the 0.6x-0.8x range or the 0.9x-1.4x range. There is no way to tell which times are likely to be good or bad.

**Comment by theRTC204** [comment]
> Thanks for taking the time to think about this.
> 
> > if you end up with a sell price in Monday AM that is anything 99 or above, you have a 90.2% chance of landing a Fluctuating pattern. At that point, it is not really any longer a question of which pattern you will get
> 
> A user of the site having just entered their data will not yet know the answer to that question. The only way they can currently get it is to look down the entirety of the giant table. At this size the table is hard to read and it is not even clear that the 90.2% is an aggregate probability or what it represents. If the user doesn't already know what they're looking for I think there's a good chance they won't be able to learn anything from the table.
> 
> > why would you not want to see every possible version, so you can make the best decision.
> 
> Because:
> 
> 1. You're looking at a giant wall of numbers. I have some limited interest in viewing such things. I think most others have much less.
> 2. There is no decision to be made at this stage of the week. Assuming you are on the fluctuating pattern, any price could be in the 0.6x-0.8x range or the 0.9x-1.4x range. There is no way to tell which times are likely to be good or bad.

> A user of the site having just entered their data will not yet know the answer to that question. The only way they can currently get it is to look down the entirety of the giant table. At this size the table is hard to read and it is not even clear that the 90.2% is an aggregate probability or what it represents.

Based on this statement, it sounds to me that the real solution here is making the high level summary more obvious, so that new users aren't required to look at the chart.

I think the previously mentioned pie chart of aggregate pattern probability will address that, which will happen as part of another issue.

> If the user doesn't already know what they're looking for I think there's a good chance they won't be able to learn anything from the table.

How does the user learn if they don't have all the facts? When you're studying or researching something, it often doesn't make sense in the beginning, but through exposure and repetition it becomes more clear.

My opinion is, by hiding valuable data you're risking confusing the user for much longer as in addition to not initially understanding the numbers, they also won't understand why patterns that previously didn't exist are suddenly appearing later on.

Even if you don't fully understand what the numbers mean, it's much simpler to grasp the concept that as you enter more data, the possible results are narrowed down - the same can't be said for new results appearing that weren't previously there.

> 1. You're looking at a giant wall of numbers. I have some limited interest in viewing such things. I think most others have much less.

This is probably true, and it's also the reason we added the visual graph. If you don't care about the detailed breakdown, you've got a chart that visualizes the results for you.

> 2. There is no decision to be made at this stage of the week. Assuming you are on the fluctuating pattern, any price could be in the 0.6x-0.8x range or the 0.9x-1.4x range. There is no way to tell which times are likely to be good or bad.

You could say there's no decision to make, because your peak won't come until Tuesday AM - if there's no decision, then there's no point to looking at the table, at which point what value does removing results from the table provide?

**Comment by doopl** [comment]
> Based on this statement, it sounds to me that the real solution here is making the high level summary more obvious, so that new users aren't required to look at the chart.

> I think the previously mentioned pie chart of aggregate pattern probability will address that, which will happen as part of another issue.

Agreed. There's no point making changes to the table for the benefit of users who just want key bits of information if the table would continue to serve them poorly.

> How does the user learn if they don't have all the facts?

By having the important facts presented in a clear and concise manner. I wasn't really talking about the guy who wants to study the intricacies of the "high, decreasing, high, decreasing, high" pattern though. For him it could be helpful to have all 56 permutations listed out. This guy was me a few weeks ago btw, lol. Incidentally it would be a bit harder for me to learn from the table now because of the extra rows of fudge factor jank.

> My opinion is, by hiding valuable data you're risking confusing the user for much longer as in addition to not initially understanding the numbers, they also won't understand why patterns that previously didn't exist are suddenly appearing later on.

> Even if you don't fully understand what the numbers mean, it's much simpler to grasp the concept that as you enter more data, the possible results are narrowed down - the same can't be said for new results appearing that weren't previously there.

Agreed, patterns appearing from nowhere is likely to cause confusion when it is noticed. If you happen to be gesturing towards my previous suggestion in #187 I'll say that there are ways to present a single aggregated row that would make it clear that it represents multiple possible patterns, and by which to ensure that to provenance of any newly displayed rows is known to the user.

Thanks again for yor replies, Ryan. I hope you're having a good day!

**Comment by nbernabe09** [comment]
If I could revitalize an idea that was offered in Issue #72, adding a quick snapshot to summarize all the current probabilities without removing the offered detail chart could simplify a lot of information overload issues that a user could encounter.

![image](https://user-images.githubusercontent.com/28968121/80139625-97971300-855b-11ea-8936-4dfbc34fb894.png)

I use this all the time on my own browser and it helps me simplify my predictions early on without looking into the details that become more valuable later within the week.
